app/src/main/java/com/example/jeff/newmovie/MainMovie.java

```
@@ -0,0 +1,144 @@
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.main_movie_activity);
+        gvMovies = (GridView)findViewById(R.id.gvMovie);
+        new FetchMovieData().execute("http://api.themoviedb.org/3/discover/movie?api_key=c8e968bcd558585cd3f858dda44ae89d");
+
+    }
+
+
+
+
+    public class FetchMovieData extends AsyncTask<String,Void,String>{
+
+        private  String LOG_TAG = "MOVIE JSON";
+        private String jsonData;
+        private final String IMAGE_URL = "http://image.tmdb.org/t/p/w300/";
+
+        @Override
+        protected String doInBackground(String... params) {
+
+
+            BufferedReader reader = null;
+            HttpURLConnection connection = null;
+
+            try {
+                URL url = new URL(params[0]);
+                connection = (HttpURLConnection) url.openConnection();
+                connection.setRequestMethod("GET");
+                connection.connect();
+
+
+
+            } catch (MalformedURLException e) {
+                e.printStackTrace();
+            } catch (IOException e) {
+                e.printStackTrace();
+            } finally {
+                if (connection == null){
+                    connection.disconnect();
+                }
+            }
+            InputStream stream = null;
+            try {
+                stream = connection.getInputStream();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+
+            reader = new BufferedReader(new InputStreamReader(stream));
+
+            StringBuffer stringBuffer = new StringBuffer();
+            String line = "";
+
+            try {
+                while ((line = reader.readLine()) != null) {
+
+                    stringBuffer.append(line +   "/n");
+
+                    jsonData = stringBuffer.toString();
+                    Log.v(LOG_TAG, jsonData);
+
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
```

Di dapat swallow yun exception - dapat irethrow (throw e)